File: Python/Problem 017.py
ones = {1: 'one', 2: 'two', 3: 'three', 4: 'four', 5: 'five', 6: 'six', 7: 'seven', 8: 'eight', 9: 'nine'}
tens_1x = {10: 'ten', 11: 'eleven', 12: 'twelve', 13: 'thirteen', 14: 'fourteen', 15: 'fifteen', 16: 'sixteen',
           17: 'seventeen',
           18: 'eighteen', 19: 'nineteen'}
tens = {2: 'twenty', 3: 'thirty', 4: 'forty', 5: 'fifty', 6: 'sixty', 7: 'seventy', 8: 'eighty', 9: 'ninety'}

letters = 0

for x in range(1, 10):
    letters += len(ones[x])                     # 1 - 9

for x in range(10, 20):
    letters += len(tens_1x[x])                  # 10 - 19

for x in range(20, 100):
    letters += len(tens[int(str(x)[0])])        # tens
    if str(x)[1] != '0':
        letters += len(ones[int(str(x)[1])])    # ones

for x in range(100, 1000):
    letters += len(ones[int(str(x)[0])]) + 7    # hundreds + 'hundred'
    if str(x)[1] != '0' or str(x)[2] != '0':
        letters += 3                            # 'and'
        if str(x)[1] == '1':
            letters += len(tens_1x[int(str(x)[1:3])])    # 10 - 19
        else:
            if str(x)[2] != '0':
                letters += len(ones[int(str(x)[2])])    # ones
            if 1 < int(str(x)[1]) <= 9:
                letters += len(tens[int(str(x)[1])])    # tens

letters += len('onethousand')

print(letters)

File: Java/Problem_017.java
package ProjectEuler;

public class Problem_017 {

    public static void main(String[] args) {
        double start = System.nanoTime();

        String[] ones = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        String[] tens_1x = {"ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
        String[] tens = {"twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};

        long letters = 0;

        // 1 - 9
        for (int i = 1; i < 10; i++) {
            letters += ones[i - 1].length();
        }

        // 10 - 19
        for (int i = 10; i < 20; i++) {
            letters += tens_1x[i - 10].length();
        }

        // 20 - 99
        for (int i = 20; i < 100; i++) {
            letters += tens[i / 10 - 2].length();
            if (i % 10 != 0) {
                letters += ones[i % 10 - 1].length();
            }
        }

        // 100 - 999
        for (int i = 100; i < 1000; i++) {
            letters += ones[i / 100 - 1].length() + "hundred".length();
            if (i % 100 != 0) {
                letters += "and".length();
                if (i % 100 / 10 == 1) {
                    letters += tens_1x[i % 10].length();
                } else {
                    if (i % 100 / 10 != 0) {
                        letters += tens[i % 100 / 10 - 2].length();
                    }
                    if (i % 10 != 0) {
                        letters += ones[i % 10 -1].length();
                    }
                }
            }
        }
        
        letters += "onethousand".length();
        System.out.println(letters);

        System.out.println((System.nanoTime() - start) / 1_000_000 + " ms");
    }

}